### PR TITLE
Allow for nullable function args

### PR DESCRIPTION
**Pull request by Pikachu920**
Target Minecraft versions: Any
Requirements: None
Related issues: #1089

Description:
Changing this behavior outright could break a lot of scripts, and as far as I can tell it was intended behavior. Also, I think that most scripters don't want to deal with null values in functions. So, I made this PR which adds the following functionality
```
function anotherTest(superString: string):
  broadcast "%{_superString}%"

function test(superString: string?):
  broadcast "%{_superString}%"

command show:
  trigger:
    test({_not set}) #broadcasts <none>
    anotherTest({_not set}) #function is never run
```

### CODE REVIEW

**Comment by Snow-Pyon** [comment]
How would this break scripts? I'm confident this behavior caused no more than pointless null checks before calling the function, something Skript shouldn't be enforcing. This particular functionality is something I'd like to be removed even if it breaks some scripts.

It is highly questionable that someone actually took approach of this weird behavior.

**Comment by Pikachu920** [comment]
Because it's been in Skript intentional or not and I've designed some things around it myself

**Comment by Syst3ms** [comment]
Uhh, nullable types ? Talking about a strong type system, it's the epitome of that. 
If you're adding these to functions, why not add them for everything ?
And why not just remove the null check ?

**Comment by Pikachu920** [comment]
>nullable types?
I only used the world nullable because it functions similar to what the annotation means in java

>If you're adding these to functions, why not add them for everything ?
like?

>And why not just remove the null check ?
where

**Comment by Syst3ms** [comment]
Why couldn't you just remove the argument null check altogether and remove these nullable types ? I seriously doubt any scripts rely on this, and even if they did, it would be so few as to where it would be neglectable.

**Comment by HYPExMon5ter** [comment]
Agreed, @bensku what’s your opinion? After all, it’s his choice

**Comment by bensku** [comment]
Functions not running when a null argument is provided makes no sense. That is not how null safety works: silently not executing a part of code is horrible, horrible idea. All functions should execute no matter if their parameters exist or not.

Null safety for scripts should be done in parser level, by verifying that a function is not called with something that could be null and producing parse error if it happens. Also, it would have to extend to variables and all expressions - including ones provided by addons.

In short, no. I would just allow null arguments for all functions and then consider if a more general-purpose nullness support should be implemented.

**Comment by Snow-Pyon** [comment]
Well, that said, I'm gonna deny this PR because the majority agrees with the fact that we shouldn't keep this behavior. I'll just remove the null check when I get time.
